lib/new_relic/agent/instrumentation/mongo.rb

```diff
@@ -16,7 +16,44 @@
 
   executes do
     NewRelic::Agent.logger.info 'Installing Mongo instrumentation'
-    install_mongo_instrumentation
+    if NewRelic::Agent::Datastores::Mongo.is_monitoring_enabled?
+      install_mongo_command_subscriber
+    else
+      install_mongo_instrumentation
+    end
+  end
+
+  class CommandSubscriber
+
+    def started(event)
+      operations[event.operation_id] = event
```

Comment:
The completed event does not have the command that was executed, so it does need it. We guarantee that there will always be a completed or failed event for every started event that fires. The only exception would be an application crash, in which in this case would not matter.